config/phobos.yml.example

```diff
@@ -109,11 +109,8 @@ listeners:
     # Possible values: [`message`, `batch` (default)]
     #  - `message` will yield individual messages from Ruby Kafka using `each_message` and will commit/heartbeat at every consumed message.
     #     This is overall a bit slower than using batch, but easier to configure.
-    #  - `batch` will yield batches from Ruby Kafka using `each_batch`, and commit/heartbeat at every consumed batch.
-    #     Due to implementation in Ruby Kafka, it should be noted that when configuring large batch sizes in combination with taking long time to consume messages,
-    #     your consumers might get kicked from the Kafka cluster for not sending a heartbeat within the expected interval.
-    #     Take this into consideration when determining your configuration. You can send heartbeats manually while
-    #     processing your messages if necessary.
+    #  - `batch` will yield batches from Ruby Kafka using `each_batch`, and commit at every consumed batch. It will
+    #     still heartbeat after every message if necessary (using the heartbeat_interval, below).
```

Comment:
I don't see a problem changing this either, but just out of curiosity - for your particular problem did you analyse if you indeed have too large batch sizes or not? I didn't see any references to this in your analysis. In general it feels like good recommendation to decrease batch size with respect to how long processing each individual message takes.

I can recommend https://github.com/phobos/phobos_prometheus or something similar to get a good picture on how your consumption is doing.

Reply:
@klippx Yep, decreasing batch size usually helps, but we've run into issues when we have a sudden spike in unexpected messages on a previously low-traffic topic. This is essentially a catch-all to make sure that batch size would only affect the network traffic for fetches and committing offsets and wouldn't risk losing the session entirely. This can end up being a fatal error as the consumer keeps consuming the same too-large batch over and over again.